central/telemetry/centralclient/client.go

```diff
@@ -134,30 +147,46 @@ func getInstanceId(ids installationDS.Store) (string, error) {
 	return ii.Id, nil
 }
 
+// initializeClient performs the common initialization logic for telemetry clients.
+// It returns nil if offline mode is enabled, or a configured client otherwise.
+func initializeClient(factory func(string) *centralClient) *centralClient {
+	if env.OfflineModeEnv.BooleanSetting() {
+		return nil
+	}
+
+	utils.Must(permanentTelemetryCampaign.Compile())
+
+	cfg := factory("")
+
+	if !cfg.IsActive() || cfg.Reload() != nil {
+		return cfg
+	}
+
+	props := getCentralDeploymentProperties()
+	cfg.Gatherer().AddGatherer(func(ctx context.Context) (map[string]any, error) {
+		return props, nil
+	})
+
+	return cfg
+}
+
 // Singleton instance collects the central instance telemetry configuration from
 // central Deployment labels and environment variables, installation store and
 // orchestrator properties. The collected data is used for configuring the
 // telemetry client. Returns nil if data collection is disabled.
 func Singleton() *centralClient {
 	once.Do(func() {
-		if env.OfflineModeEnv.BooleanSetting() {
+		// Disable telemetry when no storage key is configured
+		if env.TelemetryStorageKey.Setting() == "" {
+			client = &centralClient{Client: &phonehome.Client{}}
```

Comment:
Telemetry should not be disabled if no key is configured: the key will be set by the "runtime configuration" dynamically.

Reply:
Should it be then `if env.TelemetryStorageKey.Setting() == "" {` or the if should completely dropped?

Reply:
Please see a regrouped for clarity logic in the PR https://github.com/stackrox/stackrox/pull/15615/files#diff-deb58b2be496c7d878e57428887960c8e2fbba5b07fa039859c86fdbda30acf5R43-R49

In a general case, the client is no-op if either of the following is true:
1. Storage key is DISABLED
2. Storage key is empty AND (the code is running a non-release binary or run by go test)

On top of that, the central backend client adds:
3. Central is running in an Offline mode
4. Instance ID computation error